File: leetcode/easy/kth-distinct-string-in-an-array.py
# https://leetcode.com/problems/kth-distinct-string-in-an-array/

class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        freq = Counter(arr)
        for s in arr:
            if freq[s] == 1:
                k -= 1
            if k == 0:
                return s
        return ''

File: leetcode/easy/kth-distinct-string-in-an-array.ts
// https://leetcode.com/problems/kth-distinct-string-in-an-array/

// MINE
function kthDistinct(arr: string[], k: number): string {
  const map = new Map()
  for (let s of arr) map.set(s, map.get(s) + 1 || 1)
  for (let s of arr) if (map.get(s) === 1 && --k === 0) return s
  return ''
}

// ANSWER
